src/Parser/Mage/Fire/Modules/Features/HeatingUp.js

```diff
@@ -0,0 +1,163 @@
+];
+
+class HeatingUp extends Analyzer {
+  static dependencies = {
+    combatants: Combatants,
+    abilityTracker: AbilityTracker,
+    enemies: EnemyInstances,
+  };
+
+  fireBlastWithoutHeatingUp = 0;
+  phoenixFlamesWithoutHeatingUp = 0;
+  fireBlastWithHotStreak = 0;
+  phoenixFlamesWithHotStreak = 0;
+  targetId = 0;
+
+  on_byPlayer_cast(event) {
+    const spellId = event.ability.guid;
+    if (spellId !== SPELLS.PHOENIXS_FLAMES.id) {
+      return;
+    }
+    this.targetId = encodeTargetString(event.targetID, event.targetInstance);
+  }
+
+  on_byPlayer_damage(event) {
+    const spellId = event.ability.guid;
+    const damageTarget = encodeTargetString(event.targetID, event.targetInstance);
+    if (!GUARANTEE_CRIT_SPELLS.includes(spellId) || (spellId === SPELLS.PHOENIXS_FLAMES.id && (this.targetId !== damageTarget))) {
+      return;
+    }
+    const hasHeatingUp = this.combatants.selected.hasBuff(SPELLS.HEATING_UP.id);
+    const hasHotStreak = this.combatants.selected.hasBuff(SPELLS.HOT_STREAK.id);
+
+    switch (spellId) {
+      case SPELLS.FIRE_BLAST.id:
+        if (hasHotStreak) {
+          this.fireBlastWithHotStreak += 1;
+          debug && console.log("Fire Blast with Hot Streak @ " + formatMilliseconds(event.timestamp - this.owner.fight.start_time));
+        }
+        if (!hasHeatingUp && !hasHotStreak) {
+          this.fireBlastWithoutHeatingUp += 1;
+          debug && console.log("Fire Blast without Heating Up @ " + formatMilliseconds(event.timestamp - this.owner.fight.start_time));
+        }
+        break;
+      case SPELLS.PHOENIXS_FLAMES.id:
+        if (hasHotStreak) {
+          this.phoenixFlamesWithHotStreak += 1;
+          debug && console.log("Phoenix Flames with Hot Streak @ " + formatMilliseconds(event.timestamp - this.owner.fight.start_time));
+        }
+        if (!hasHeatingUp && !hasHotStreak) {
+          this.phoenixFlamesWithoutHeatingUp += 1;
+          debug && console.log("Phoenix Flames without Heating Up @ " + formatMilliseconds(event.timestamp - this.owner.fight.start_time));
+        }
+        break;
+      default:
+        return;
+    }
+  }
+
+  get fireBlastWasted() {
+    return this.fireBlastWithoutHeatingUp + this.fireBlastWithHotStreak;
+  }
+
+  get phoenixFlamesWasted() {
+    return this.phoenixFlamesWithoutHeatingUp + this.phoenixFlamesWithHotStreak;
+  }
+
+  get totalWasted() {
+    return this.fireBlastWasted + this.phoenixFlamesWasted;
+  }
+
+  get fireBlastUtil() {
+    return 1 - this.fireBlastMissedPercent;
+  }
+
+  get phoenixFlamesUtil() {
+    return 1 - this.phoenixFlamesMissedPercent;
+  }
+
+  get fireBlastMissedPercent() {
+    return this.fireBlastWasted / this.abilityTracker.getAbility(SPELLS.FIRE_BLAST.id).casts;
+  }
+
+  get phoenixFlamesMissedPercent() {
+    return this.phoenixFlamesWasted / this.abilityTracker.getAbility(SPELLS.PHOENIXS_FLAMES.id).casts;
+  }
+
+  suggestions(when) {
+		when(this.fireBlastMissedPercent).isGreaterThan(0.1)
```

Comment:
You should use the new suggestion thresholds format even if this isn't going in the checklist.